Yoast/comment-hacks
# Update to YoastCS 1.3.0

:warning: The `grunt` `phpcs` command, or rather then whole `shell` command should be disabled, but it is unclear to me how to do this. Help appreciated.

The `grunt` [`shell` command](https://github.com/Yoast/plugin-grunt-tasks#the-shell-task) does linting of the PHP files + a PHPCS check. This should not be done via grunt and the Travis script already contains a PHP lint command and with this PR a PHPCS check.

---


This PR will be easiest to review by looking at each commit individually.

### Composer: use YoastCS 1.3.0

... and update and add relevant Composer scripts.

Note: only the relevant PHPCS/YoastCS dependencies have been updated, the other dependencies have not been touched.

### Travis: add CS check

### CI/QA: rename PHPCS configuration file

Using a `.dist` extension allows for individual developers to overload the  `phpcs.xml.dist` with a local `phpcs.xml` file. This file can include the `dist` file  as a base ruleset using `<rule ref="./.phpcs.xml.dist"/>`.

To facilitate this, the typical file names for the overloaded PHPCS config file have been added to the `.gitignore` file.

### Build/PHPCS: set up the PHPCS ruleset properly

* Add XSD schema.
* Set the command line arguments.
* Remove `exclude-pattern`s which are already set via YoastCS.
* Add configuration for the `WordPress.WP.I18n` sniff.
* Add configuration for the `Yoast.Files.Filename` sniff and remove the exclusion.
* Add configuration for the `WordPress.NamingConventions.PrefixAllGlobals` sniff.

### Build/PHPCS: add temporary exclusion

There is one type of issue remaining for which a technical debt issue has been created - #50 -.
For now, ignore these issues.

### PHPCS: replace old-style ignore comment with new style

This code is only used when `WP_DEBUG` is on to display debugging information.

Includes reformatting the surrounding function call for readability.

### PHPCS: whitelist a used parameter

### PHPCS: add reason to whitelist comment 


## Testing

* Check the detailed Travis script output for the build with the `PHPCS=1` environment variable.
* Locally
    - Check out this branch.
    - Run the following commands:
        - `composer install`
        - `composer check-cs`
        - See a clean run without any errors or warnings.
